flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -1313,13 +1274,26 @@ struct EmboxCommonConversion : public FIROpConversion<OP> {
       auto ty = boxEleTy.cast<fir::CplxType>();
       return doComplex(getKindMap().getRealBitsize(ty.getFKind()));
     }
-    if (auto ty = boxEleTy.dyn_cast<fir::CharacterType>()) {
-      TODO();
-    }
-    if (auto ty = boxEleTy.dyn_cast<fir::LogicalType>())
+    if (auto ty = boxEleTy.dyn_cast<fir::CharacterType>())
+      return doCharacter(getKindMap().getCharacterBitsize(ty.getFKind()),
+                         ty.getLen());
+    if (auto ty = boxEleTy.dyn_cast<fir::LogicalType>()) {
+      // TODO: doesn't the runtime need to know these are LOGICAL? Pretend they
+      // are INTEGER for now.
       return doInteger(getKindMap().getLogicalBitsize(ty.getFKind()));
-    if (auto seqTy = boxEleTy.dyn_cast<fir::SequenceType>())
+    }
+    if (auto seqTy = boxEleTy.dyn_cast<fir::SequenceType>()) {
+      if (auto charTy = seqTy.getEleTy().dyn_cast<fir::CharacterType>()) {
+        // TODO: assumes the row is the length of the CHARACTER. This is true by
+        // construction, but it may not hold after optimizations have run.
```

Comment:
Can optimization change array shapes ?

Reply:
We might want to linearize an array, for example.